From 2ddfab3a95edc34d231d900a0c8f98a8bcb80227 Mon Sep 17 00:00:00 2001
From: Chris Lalancette <clalance@redhat.com>
Date: Fri, 10 Jun 2011 17:16:06 -0400
Subject: [PATCH 07/13] Move create_failure into dcloudgahp_commands.cpp.

It never quite sat well with me that create_success was
in dcloudgahp_commands.cpp while create_failure was in
dcloudgahp_common.cpp.  It turns out that only one
place outside of dcloudgahp_commands.cpp was using it,
and using it very simply at that.  Just move it into
dcloudgahp_commands.cpp statically, and replace the
one other use with a hardcoded string.

Signed-off-by: Chris Lalancette <clalance@redhat.com>
---
 src/deltacloud_gahp/dcloudgahp_commands.cpp |   29 +++++++++++++++++++++++++++
 src/deltacloud_gahp/dcloudgahp_common.cpp   |   28 --------------------------
 src/deltacloud_gahp/dcloudgahp_common.h     |    2 -
 src/deltacloud_gahp/dcloudgahp_main.cpp     |    2 +-
 4 files changed, 30 insertions(+), 31 deletions(-)

diff --git a/src/deltacloud_gahp/dcloudgahp_commands.cpp b/src/deltacloud_gahp/dcloudgahp_commands.cpp
index 774d5a3..1cf304b 100644
--- a/src/deltacloud_gahp/dcloudgahp_commands.cpp
+++ b/src/deltacloud_gahp/dcloudgahp_commands.cpp
@@ -7,6 +7,7 @@
 #include <errno.h>
 #include <fcntl.h>
 #include <unistd.h>
+#include <stdarg.h>
 #include <ctype.h>
 #include <sys/types.h>
 #include <sys/stat.h>
@@ -134,6 +135,34 @@ cleanup:
     return password;
 }
 
+static std::string create_failure(const char *req_id, const char *err_msg, ...)
+{
+    std::string buffer;
+    va_list ap;
+    char *tmp;
+    unsigned int i;
+
+    buffer += req_id;
+    buffer += ' ';
+
+    va_start(ap, err_msg);
+    vasprintf(&tmp, err_msg, ap);
+    va_end(ap);
+
+    for (i = 0; i < strlen(tmp); i++) {
+        if (tmp[i] == ' ')
+            buffer += '\\';
+        buffer += tmp[i];
+    }
+    free(tmp);
+
+    buffer += '\n';
+
+    dcloudprintf(buffer.c_str());
+
+    return buffer;
+}
+
 static std::string create_instance_output(char * reqid,
                                           struct deltacloud_instance *inst)
 {
diff --git a/src/deltacloud_gahp/dcloudgahp_common.cpp b/src/deltacloud_gahp/dcloudgahp_common.cpp
index 8100725..f3c0998 100644
--- a/src/deltacloud_gahp/dcloudgahp_common.cpp
+++ b/src/deltacloud_gahp/dcloudgahp_common.cpp
@@ -22,34 +22,6 @@ void dcloudprintf_internal(const char *function, const char *fmt, ...)
     }
 }
 
-std::string create_failure(const char *req_id, const char *err_msg, ...)
-{
-    std::string buffer;
-    va_list ap;
-    char *tmp;
-    unsigned int i;
-
-    buffer += req_id;
-    buffer += ' ';
-
-    va_start(ap, err_msg);
-    vasprintf(&tmp, err_msg, ap);
-    va_end(ap);
-
-    for (i = 0; i < strlen(tmp); i++) {
-        if (tmp[i] == ' ')
-            buffer += '\\';
-        buffer += tmp[i];
-    }
-    free(tmp);
-
-    buffer += '\n';
-
-    dcloudprintf(buffer.c_str());
-
-    return buffer;
-}
-
 Gahp_Args::Gahp_Args()
 {
     argv = NULL;
diff --git a/src/deltacloud_gahp/dcloudgahp_common.h b/src/deltacloud_gahp/dcloudgahp_common.h
index 0026fba..4fa443b 100644
--- a/src/deltacloud_gahp/dcloudgahp_common.h
+++ b/src/deltacloud_gahp/dcloudgahp_common.h
@@ -16,8 +16,6 @@ void dcloudprintf_internal(const char *function, const char *fmt, ...);
 
 extern FILE *logfp;
 
-std::string create_failure(const char *req_id, const char *err_msg, ...);
-
 /* Users of GahpArgs should not manipulate the class data members directly.
  * Changing the object should only be done via the member functions.
  * If argc is 0, then the value of argv is undefined. If argc > 0, then
diff --git a/src/deltacloud_gahp/dcloudgahp_main.cpp b/src/deltacloud_gahp/dcloudgahp_main.cpp
index 03deabb..0cad719 100644
--- a/src/deltacloud_gahp/dcloudgahp_main.cpp
+++ b/src/deltacloud_gahp/dcloudgahp_main.cpp
@@ -281,7 +281,7 @@ static void *worker_function(void *ptr)
          * earlier, so there is no reason we can't parse it again.
          */
         dcloudprintf("Failed to parse command again\n");
-        output_string = create_failure("0", "Command_Parse_Failure");
+        output_string = "0 Command_Parse_Failure\n";
         goto cleanup;
     }
 
-- 
1.7.4.4

